tools/osx-codesign.sh

```diff
@@ -1,17 +1,18 @@
-#!/bin/bash
+#!/bin/sh
 
 set -x
 set -e
 
-if [ "X$SIGN" == "X" ]; then
-  echo "No SIGN environment var.  Skipping codesign." >&2
+# shellcheck disable=SC2154
+[ -z "$SIGN" ] && \
+  echo "No SIGN environment var.  Skipping codesign." >&2 && \
   exit 0
-fi
 
 # All macOS executable binaries in the bundle must be codesigned with the
 # hardened runtime enabled.
 # See https://github.com/nodejs/node/pull/31459
 
+# shellcheck disable=SC2154
```

Comment:
Rather than this disable directive, would it be better to add a `-z "$PKGDIR"` check similar to that on line 7 for `$SIGN`? Both are supplied by the Makefile and guaranteed to be not-empty. So it would seem to me that we should either check for both or assume non-empty for both. But we seem to check for only one. Check for both?

Or is that a modification outside the scope of this PR and should be done later?